core/cc/dl_loader.cpp

```diff
@@ -51,6 +54,16 @@ void* load(const char* name) {
         }
     }
     return res;
+#elif TARGET_OS == GAPID_OS_LINUX
+    void* res = dlopen(name, RTLD_NOW | RTLD_LOCAL);
+    if (res != nullptr) {
+        return res;
+    }
+    // Try name + ".1"
+    size_t len = strlen(name) + sizeof(".1");
+    char* fallback_name = new char[len];
+    strcat(fallback_name, ".1");
```

Comment:
1. You need to clear the memory.
2. You need to actually copy in name.
3. You need to free the memory.
4. You're better of using `std::stringstream`